File: 90-SubsetsII.py
class Solution(object):
    def subsetsWithDup(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        nums.sort()
        res = []
        n = len(nums)
        def dfs(val, start):
            res.append(val)
            for i in range(start, n):
                if i>start and nums[i] == nums[i-1]:
                    continue
                dfs(val+[nums[i]], i+1)
                
        dfs([],0)
        return res

File: JavaScript/90-Subsets II.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var subsetsWithDup = function(nums) {
    let res = new Array();
    nums.sort(function(a,b){return a-b;});
    
    let dfs = (start, val)=>{
        res.push(val);
        for (let i=start; i<nums.length; i++){
            if (i > start && nums[i] === nums[i-1]) continue;
            dfs(i+1, val.concat([nums[i]]));
        }
    }
    dfs(0, new Array());
    return res;
};